staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -132,24 +131,39 @@ func NonSlidingUntilWithContext(ctx context.Context, f func(context.Context), pe
 // Close stopCh to stop. f may not be invoked if stop channel is already
 // closed. Pass NeverStop to if you don't want it stop.
 func JitterUntil(f func(), period time.Duration, jitterFactor float64, sliding bool, stopCh <-chan struct{}) {
-	BackoffUntil(f, NewJitteredBackoffManager(period, jitterFactor, &clock.RealClock{}), sliding, stopCh)
+	b := Backoff{Duration: period, Jitter: jitterFactor}
+	BackoffUntil(f, RealTimer, b.Step, sliding, stopCh)
 }
 
+var (
+	// RealTimer can be passed to methods that need a TimerFunc.
+	RealTimer = clock.RealClock{}.NewTimer
+)
+
+// TimerFunc allows the caller to inject a timer for testing. Most callers should use RealTimer.
+type TimerFunc func(time.Duration) clock.Timer
+
 // BackoffUntil loops until stop channel is closed, run f every duration given by BackoffManager.
 //
 // If sliding is true, the period is computed after f runs. If it is false then
 // period includes the runtime for f.
-func BackoffUntil(f func(), backoff BackoffManager, sliding bool, stopCh <-chan struct{}) {
```

Comment:
this change looks less disruptive https://grep.app/search?q=wait.BackoffUntil&filter[lang][0]=Go

Reply:
The open question with me deprecating instead of removing `New*` is whether we need to add a transition method on the existing manager.  If we do, users of wait.BackoffUntil have to change twice in the future (once now to use that helper and then once later when the constructors are removed).  If we don't add the helper, they have to change now but only do so once.  Both are disruptive.  Changing twice makes this easier to land but more work for them overall.

Preference?

Reply:
I added the helper and pushed that change.  I'll note that callers can choose to use that until the manager is deprecated.